src/hostcalls.rs

```diff
@@ -678,6 +678,66 @@ pub fn done() -> Result<(), Status> {
     }
 }
 
+extern "C" {
+    fn proxy_define_metric(
+        metric_type: MetricType,
+        name_data: *const u8,
+        name_size: usize,
+        return_id: *mut u32,
+    ) -> Status;
+}
+
+pub fn define_metric(metric_type: MetricType, name: &str) -> Result<u32, Status> {
+    let mut return_id: u32 = 0;
+    unsafe {
+        match proxy_define_metric(metric_type, name.as_ptr(), name.len(), &mut return_id) {
+            Status::Ok => Ok(return_id),
+            status => panic!("unexpected status: {}", status as u32),
+        }
+    }
+}
+
+extern "C" {
+    fn proxy_get_metric(metric_id: u32, return_value: *mut u64) -> Status;
+}
+
+pub fn get_metric(metric_id: u32) -> Result<u64, Status> {
+    let mut return_value: u64 = 0;
+    unsafe {
+        match proxy_get_metric(metric_id, &mut return_value) {
+            Status::Ok => Ok(return_value),
+            Status::BadArgument => Err(Status::BadArgument),
+            status => panic!("unexpected status: {}", status as u32),
```

Comment:
I'm looking into implementation of this callback on Envoy side (https://github.com/envoyproxy/envoy-wasm/blob/master/source/extensions/common/wasm/context.cc#L1250-L1268).

It can return one of `Ok / BadArgument / NotFound`.

Why only `BadArgument` is handled here but not `NotFound` ?

Reply:
I was playing around with a higher-level object wrappers to metrics, where `metric_id` is never exposed to the consumer, where `NotFound` couldn't happen, and I forgot to add it to the `hostcalls`.

Thanks for catching it!